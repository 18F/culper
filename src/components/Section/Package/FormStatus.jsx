import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import { Svg } from 'components/Form'
import AnimateReviewIcon from './AnimateReviewIcon'
import AnimateCheckmarkIcon from './AnimateCheckmarkIcon'

const FormStatus = ({
  progressWidth, isValid, isTransitioning, onTransitionEnd,
}) => (
  <div className="progress-container">
    <div className="review-icon">
      {isTransitioning ? (
        <img src="/img/review-checking.svg" alt="" />
      ) : (
        <AnimateReviewIcon valid={isValid} />
      )}
    </div>
    <div className="progress-outline">
      <div className="progress-default">
        <div
          className={classnames('progress', { transition: isTransitioning })}
          style={{ width: `${progressWidth}%` }}
          onTransitionEnd={onTransitionEnd}
        />
      </div>
    </div>
    <div className="icon-container">
      <span className="icon">
        {isTransitioning ? (
          <Svg src="/img/checkmark.svg" />
        ) : (
          <AnimateCheckmarkIcon valid={isValid} />
        )}
      </span>
    </div>
  </div>
)

FormStatus.propTypes = {
  progressWidth: PropTypes.number,
  isValid: PropTypes.bool,
  isTransitioning: PropTypes.bool,
  onTransitionEnd: PropTypes.func,
}

FormStatus.defaultProps = {
  progressWidth: 100,
  isValid: false,
  isTransitioning: false,
  onTransitionEnd: () => {},
}

export default FormStatus
