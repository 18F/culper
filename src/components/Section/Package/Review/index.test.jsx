import React from 'react'
import { mount } from 'enzyme'

import { PackageReview } from './index'

import FormStatus from '../FormStatus'

describe('The PackageReview component', () => {
  const component = mount(<PackageReview />)

  it('renders without errors', () => {
    expect(component.exists()).toBe(true)
  })

  it('renders the FormStatus component', () => {
    const formStatus = component.find(FormStatus)
    expect(formStatus.length).toBe(1)
    expect(formStatus.prop('isTransitioning')).toBe(true)
    expect(formStatus.prop('progressWidth')).toBe(0)

    component.setState({ width: 50 })
    expect(formStatus.prop('progressWidth')).toBe(50)
  })
})
