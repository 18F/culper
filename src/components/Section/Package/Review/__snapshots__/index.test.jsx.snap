// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PackageReview component renders without errors 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="submission-status"
  >
    <ReactMarkdown
      containerTagName="div"
      parserOptions={Object {}}
      renderers={
        Object {
          "link": [Function],
        }
      }
      source="# Verifying your information is complete
Checking your form..."
    />
    <FormStatus
      isTransitioning={true}
      isValid={false}
      onTransitionEnd={[Function]}
      progressWidth={0}
    />
  </div>,
  "nodes": Array [
    <div
      className="submission-status"
    >
      <ReactMarkdown
        containerTagName="div"
        parserOptions={Object {}}
        renderers={
          Object {
            "link": [Function],
          }
        }
        source="# Verifying your information is complete
Checking your form..."
      />
      <FormStatus
        isTransitioning={true}
        isValid={false}
        onTransitionEnd={[Function]}
        progressWidth={0}
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <PackageReview
        formIsValid={false}
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PackageReview {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "onTransitionEnd": [Function],
        "props": Object {
          "formIsValid": false,
        },
        "refs": Object {},
        "state": Object {
          "width": 0,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="submission-status"
        >
          <ReactMarkdown
            containerTagName="div"
            parserOptions={Object {}}
            renderers={
              Object {
                "link": [Function],
              }
            }
            source="# Verifying your information is complete
Checking your form..."
          />
          <FormStatus
            isTransitioning={true}
            isValid={false}
            onTransitionEnd={[Function]}
            progressWidth={0}
          />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="submission-status"
        >
          <ReactMarkdown
            containerTagName="div"
            parserOptions={Object {}}
            renderers={
              Object {
                "link": [Function],
              }
            }
            source="# Verifying your information is complete
Checking your form..."
          />
          <FormStatus
            isTransitioning={true}
            isValid={false}
            onTransitionEnd={[Function]}
            progressWidth={0}
          />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PackageReview
    formIsValid={false}
  />,
}
`;
